presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -2064,6 +2114,84 @@ public AnalyzePropertyManager getAnalyzePropertyManager()
         return analyzePropertyManager;
     }
 
+    @Override
+    public QualifiedObjectName redirectTable(Session session, QualifiedObjectName tableName, WarningCollector warningCollector)
+    {
+        requireNonNull(session, "session is null");
+        requireNonNull(tableName, "tableName is null");
+        requireNonNull(warningCollector, "warningCollector is null");
+
+        Set<QualifiedObjectName> visitedTableNames = new HashSet<>();
+        visitedTableNames.add(tableName);
+        int count = TABLE_REDIRECTION_MAX_TIMES;
+        while (count-- > 0) {
```

Comment:
We avoid using increment/decrement in an expression. This can be written as a traditional for loop
```java
for (int count = 0; count < MAX_TABLE_REDIRECTIONS; count++) {